internal/testrunner/compiler_runner.go

```diff
@@ -506,6 +508,44 @@ func (c *compilerTest) verifyUnionOrdering(t *testing.T) {
 	})
 }
 
+func (c *compilerTest) verifyParentPointers(t *testing.T) {
```

Comment:
This will end up repeating the same tests given we share source files so much.

Maybe this should just be an assertion in the test harness's GetSourceFile implementation? Or, we `SyncSet[NodeId]` this?

Then again, it's only 4% of the total runtime.

Reply:
We leave it as-is for now so someone can go and make tests 5% faster by removing a "low value test" after we're mostly done porting and nobody remembers them catching anything recently, like we did with these parent assert tests in strada. 😂

Reply:
Yeah I simply cannot judge, my union order consistency checks take more time than the parent checks